File: TP1-Java/Electronic.java
public class Electronic {
    private String idToko;
    private String productName;
    private String category;
    private int price;

    public Electronic(String idToko, String productName, String category, int price) {
        this.idToko = idToko;
        this.productName = productName;
        this.category = category;
        this.price = price;
    }

    // Getters
    public String getIdToko() {
        return idToko;
    }

    public String getProductName() {
        return productName;
    }

    public String getCategory() {
        return category;
    }

    public int getPrice() {
        return price;
    }

    // Setters
    public void setIdToko(String idToko) {
        this.idToko = idToko;
    }

    public void setProductName(String productName) {
        this.productName = productName;
    }

    public void setCategory(String category) {
        this.category = category;
    }

    public void setPrice(int price) {
        this.price = price;
    }

    // Display method
    public void printInformation() {
        System.out.println("Kode     : " + idToko);
        System.out.println("Nama     : " + productName);
        System.out.println("Kategori : " + category);
        System.out.printf("Harga    : Rp %,d\n", price);
        System.out.println("------------------------------");
    }
}

File: TP1-C++/Electronic.cpp
#include <bits/stdc++.h>
using namespace std;

class Electronic {
private:
    string id_toko;
    string product_name;
    string category;
    int price;

public:
    // Constructor
    Electronic(string id, string name, string cat, int prc) {
        id_toko = id;
        product_name = name;
        category = cat;
        price = prc;
    }

    // Getters
    string getId() {
        return id_toko;
    }

    string getProductName() {
        return product_name;
    }

    string getCategory() {
        return category;
    }

    int getPrice() {
        return price;
    }

    // Setters
    void setId(string id) {
        id_toko = id;
    }

    void setProductName(string name) {
        product_name = name;
    }

    void setCategory(string cat) {
        category = cat;
    }

    void setPrice(int prc) {
        price = prc;
    }

    // Display product info
    void productInformation() {
        cout << "Kode: " << getId() << endl;
        cout << "Nama: " << getProductName() << endl;
        cout << "Kategori: " << getCategory() << endl;
        cout << "Harga: Rp " << fixed << setprecision(0) << getPrice() << endl;
        cout << string(30, '-') << endl;
    }
};
